ImageAnalysis/image_analysis/analyzers/UC_GenericLightSpecCam.py

```diff
@@ -30,7 +30,9 @@ def __init__(self,
                  calibration_0th_order_pixel: float = 1186.24,
                  minimum_wavelength_analysis: float = 150.0,
                  optimization_central_wavelength: float = 420.0,
-                 optimization_bandwidth_wavelength: float = 10.0
+                 optimization_bandwidth_wavelength: float = 10.0,
+                 spec_correction_bool: bool = True,
+                 spec_correction_files: List[str] = []
```

Comment:
Giving a mutable (such as a list or a dict) as a default for a keyword argument [can give unexpected results](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments). 
It's better to set None as default, followed by `if spec_correction_files is None: spec_correction_files = []` in the body.